src/components/ChannelPreview/ChannelPreview.tsx

```diff
@@ -82,23 +82,20 @@ export const ChannelPreview = <
     return () => client.off('notification.mark_read', handleEvent);
   }, []);
 
-  useEffect(() => {
+  const refreshUnreadCount = useCallback(() => {
     if (isActive || muted) {
       setUnread(0);
     } else {
       setUnread(channel.countUnread());
     }
-  }, [channel, isActive, muted]);
+  }, [channel, isActive, muted, channelUpdateCount]);
```

Comment:
```suggestion
  }, [channel, isActive, muted]);
```
Is this dependency needed? It's not being used within the scope of that function.

Reply:
@arnautov-anton Yes, it is needed - it is a value incremented to force rerender of the `ChannelPreview` component. It is used to notify `ChannelPreview` component that connection has been recovered. It is the actual fix that solves the problem that the component have not reacted to the `'connection.recovered'` event. By registering the the listener one level above you create only one listener for connection recovery instead of registering it in every `ChannelPreview` component.